Sources/Starknet/Accounts/StarknetAccount.swift

```diff
@@ -40,44 +40,48 @@ public class StarknetAccount: StarknetAccountProtocol {
         StarknetDeployAccountTransactionV3(signature: signature, l1ResourceBounds: params.resourceBounds.l1Gas, nonce: params.nonce, contractAddressSalt: salt, constructorCalldata: calldata, classHash: classHash, forFeeEstimation: forFeeEstimation)
     }
 
-    public func signV1(calls: [StarknetCall], params: StarknetInvokeParamsV1, forFeeEstimation: Bool) throws -> StarknetInvokeTransactionV1 {
+    public func signV1(calls: [StarknetCall], params: StarknetInvokeParamsV1, forFeeEstimation: Bool) async throws -> StarknetInvokeTransactionV1 {
```

Comment:
I'm not sure whether signing methods being async and relying on `provider.getChainId()` is a problem. Theoretically, someone might want to sign a transaction without having a properly working provider and use it somewhere else, practically, I'm not sure that's a use case the SDK is intended for. 

We could fetch `chainId` upon creation of `StarknetAccount`, but that wouldn't change much anyway.

Reply:
No, I don't think this a valid use case for this sdk either.